File: c++/MaxSubArray.cpp
class Solution
{
  public:
    int maxSubArray(vector<int> &nums)
    {
        int max = -2147483648;
        int sum = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            if (sum < 0)
            {
                sum = 0;
            }
            sum = sum + nums[i];
            if (sum > max)
            {
                max = sum;
            }
        }
        return max;
    }
};

File: python/maxSubarray.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = nums[0]
        s = 0
        for num in nums:
            if (s < 0):
                s = 0
            s += num            
                
            res = max(res, s)
            
        return res